### DISCUSSION TITLE
Did I mention that the UI/UX is REALLY REALLY good???

### DISCUSSION THREAD

**Discussion by Lojemiru** [akatori-game/feedback#121] [Feedback]
First things first: the UI/UX work here is **spectacular**. Legitimately have not seen better menus done in a game before; simultaneously super stylish and very functional. Spent way more time than I should have just flipping through the equipment menus just to watch the transition effects :D

**Demo critiques** (not bugs, bugs get issues):
- Thorn block sprite asset is stretched in many locations and it's quite noticeable. I presume this is a placeholder given the rest of the art style, but thought it worth mentioning regardless. If you want to keep the block usage I'd recommend looking at GM's built-in nineslicing, though I'm sure you don't need me to lecture you on GM features with how polished most of this felt.
- First boss feels perhaps a bit tanky for the players' combat experience and the boss dealing direct contact damage is confusing when little else has outside of specific attacks up to that point. I took a *lot* of deaths to accidentally walking or dodging into the boss, though much of that can simply be attributed to my general ineptitude at video games :)
    - Possible solution for contact damage confusion: if an enemy deals passive contact damage, make hits push the player back similar to Hollow Knight? Not sure if that inconsistency would worsen things.
- Dodging should respect input direction as opposed to animation direction when possible. There were a lot of times I tried to dash out of the end of the combo melee and went in the direction of the combo instead of away. This may have contributed to my impressions of the first boss being clunky in regard to contact damage.
- Parallax for the furthest layer in the cherry tree temple area feels way too fast; in particularly open areas I found myself a bit motion sick. I didn't look through all of the accessibility settings as it was just a few rooms, but a "reduced motion" option wouldn't be a bad consideration if you want to stick with this (or a bad consideration in general).
- The falling paper explosions' hitbox may linger a bit longer than is visually represented, or during the final few frames of the animation when it seems over. I would personally err on the side of ending a few frames early.
- Several default controller binding issues were present; they've already been reported, but if you're experiencing a lot of issues with a self-made system might I recommend [Juju Adam's Input library](https://www.jujuadams.com/Input/)? I've been applying it to one of my projects recently and it solves a *ton* of the usual issues; particularly nice is that it utilizes a copy of the SDL2 database so it has appropriate bindings and constants for practically all controllers.
- Lots of empty space just after the cart shop; I presume this will be filled out with more NPCs or such as the game progresses, but it might be worth considering sprucing up the space a bit.
- Probably wouldn't hurt to make the player hold to confirm when purchasing blessings; I could see lots of accidental unlocks from finger slips in the future.
- Room transitions could probably stand to have more of an entry fade; the immediate jump to black is a bit jarring, though the fade in on the other side looks very nice.

**Demo positives:**
- As stated above, **HOLY CRAP YOUR UI IS SO COOL.** Menus are insanely slick and carry excellent functionality and behavior communication.
- Artstyle is wonderful. There's some complaining these days about pixelart being overdone but it's specific *styles* that people are tired of, and Akatori definitely stands out as different. Love the overall palette usages and how smooth everything looks.
- Music is highly fitting for each environment and situation. Didn't get to hear the first boss track (known issue, I'm told), but the second boss' track was an ABSOLUTE JAM.
- Player movement feels really nice. Movement speed is neither too fast nor too slow, dashes are wonderful, and you've either got a generous collision box or well-tuned coyote time. Really appreciate the midair hover when landing attacks too, it feels just about perfect.
- The charm system seems really well designed. Probably too limited in demo form to fully realize how it plays out, but I really like the framing of major slot vs. minor slot abilities. I also appreciated the styling of the blades charm to contrast the projectile with an inversion/limitation of the air dodge, though I'm not sure how it'd play out long-term. Definitely the kind of thing I'd personally turn on for some bosses but mostly leave off because you can go crazy fast with air dodges.
- Second boss was a total blast to play. Visual styling, attack waves, and the shield mechanics were a ton of fun!
- Level and combat design overall seems good. I have no real complaints, just a handful of platforms that were occasionally juuust out of reach from where players would usually want to jump to them.
- I love how you can dash off a ledge, jump midair quickly, then leverage the air dash to get a *ton* of distance. Please don't remove this, it's really fun and could lead to some fun speedrun trickery :)

**Overall impression:**
Really really good. Definitely see the Hollow Knight inspiration, but it's not a bad thing by any means; you take the gameplay foundation that game set and fly with it, spiraling off into your own design :)

Super excited to see where this goes, and very grateful I got to playtest the alpha. You've got a long road ahead of you, but I fully trust that you can execute on the vision presented in this alpha.

**Comment by XenoWarrior** [comment]
Hey @Lojemiru, this feedback looks incredible. I see a bunch of potential enhancements suggested, and general good feedback!

Thank you for your time writing this, and I'm glad you enjoyed the demo!
